homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -89,6 +89,8 @@
 KEY_PIPELINE_CONVERSATION_DATA: HassKey[dict[str, PipelineConversationData]] = HassKey(
     "pipeline_conversation_data"
 )
+# Number of response chunks to handle before considering to stream the response
+STREAM_RESPONSE_CHUNKS = 15
```

Comment:
Let's replace this with an estimated count of the number of tokens in the stream. We need to accumulate the text string of the response and compute `len(text) // 4` to estimate the token count. If this exceeds 15, assume streaming.

Doing this will ensure that streaming works well enough for agents that produce chunks with varying sizes (token, word, phrase, sentence, etc.).